components/match2/commons/match_group_input_util.lua

```diff
@@ -205,29 +205,33 @@ function MatchGroupInputUtil.readOpponent(match, opponentIndex, options)
 	opponent.name = Opponent.toName(opponent)
 
 	local substitutions
-	if opponent.type == Opponent.team and Logic.isNotEmpty(opponent.name) then
+	if opponent.type == Opponent.team then
```

Comment:
Have you tested it on empty Team Inputs? `{{TeamOpponent|}}`?

Reply:
fucked up the merge conflict resolve ...

Reply:
unresolving this as it's still the same change

Reply:
opp.name is not set at that point, so if we would not remove it we would never enter the inside of the if

Reply:
No but previously there was a check if opponent.toName() did not return nil or empty str, before going into this nest. Does it have any impact of a case where toName may return an empty value?

Reply:
and `Opponent.toName` could return `nil`, if a team template is invalid, right?

Reply:
yeah but i am passing `Opponent.toName(opponent) or '',` to the `readPlayersOfTeamNew` function